src/lib.rs

```diff
@@ -808,6 +841,14 @@ impl<EF: ExtraTokenFields, TT: TokenType, TE: ErrorResponseType> Client<EF, TT,
         &self,
         code: AuthorizationCode
     ) -> Result<TokenResponse<EF, TT>, RequestTokenError<TE>> {
+        block_on(self.exchange_code_async(code))
+    }
+
+    /// async edition
+    pub fn exchange_code_async(
```

Comment:
futures trait also [changes greatly](https://rust-lang-nursery.github.io/futures-rs/) in [Rust 2018](https://rust-lang-nursery.github.io/edition-guide/rust-2018/control-flow/async-await-for-easier-concurrency.html).
How about doing this as a provisional response for the past few months?

* exchange_code_async
* exchange_password_async
* exchange_client_credentials_async
* exchange_refresh_token_async

Reply:
Alternative choice: just constructs and returns [`http::request::Request`](https://docs.rs/http/0.1.13/http/request/struct.Request.html) object.
User can chose own http request library e.g. curl, reqwest, hyper, actix-web ...etc.

Reply:
I think it's fine to put the pre-Rust 2018 async functionality into this crate. However, I think we should do this via a separate `Client` implementation (i.e., `oauth2::async::Client`). Then, when we're ready to add support for Rust 2018 async, we can call that `oauth2::async2018::Client` or something. Each of these can present the same `exchange_*` functions, just with different return types. This change will probably require some refactoring to share as much code as possible between the different `Client` implementations.